Kickstarter-iOS/Views/Controllers/CancelPledgeViewController.swift

```diff
@@ -161,12 +194,28 @@ final class CancelPledgeViewController: UIViewController {
   // MARK: - Accessors
 
   @objc private func dismissKeyboard() {
-    self.view.endEditing(true)
+    self.viewModel.inputs.textFieldShouldReturn()
   }
 
   @objc private func goBackButtonTapped() {
     self.viewModel.inputs.goBackButtonTapped()
   }
+
+  @objc private func cancelPledgeButtonTapped() {
+    self.viewModel.inputs.cancelPledgeButtonTapped()
+  }
+}
+
+extension CancelPledgeViewController: UITextFieldDelegate {
+  func textFieldDidEndEditing(_ textField: UITextField) {
+    self.viewModel.inputs.textFieldDidEndEditing(with: textField.text)
+  }
+
+  func textFieldShouldReturn(_: UITextField) -> Bool {
+    self.viewModel.inputs.textFieldShouldReturn()
```

Comment:
We could consider returning the boolean from the viewModel and writing a test for it.

Reply:
Since we aren't doing any validation here, I'm not sure it's necessary because we always return `true`. If we were to actually add validation where the result of the validation would dictate the response, then yeah for sure that would go through the view model.

Reply:
That's fair 👍